x-pack/plugins/lens/public/state_management/init_middleware/load_initial.ts

```diff
@@ -108,78 +96,39 @@ export function loadInitial(
     history?: History<unknown>;
   }
 ) {
-  const { getState, dispatch } = store;
   const { attributeService, notifications, data, dashboardFeatureFlag } = lensServices;
-  const { persistedDoc } = getState().lens;
+  const currentSessionId = data.search.session.getSessionId();
 
+  const { lens } = store.getState();
   if (
     !initialInput ||
     (attributeService.inputIsRefType(initialInput) &&
-      initialInput.savedObjectId === persistedDoc?.savedObjectId)
+      initialInput.savedObjectId === lens.persistedDoc?.savedObjectId)
   ) {
-    return initializeDatasources(
-      datasourceMap,
-      getState().lens.datasourceStates,
-      undefined,
-      initialContext,
-      {
-        isFullEditor: true,
-      }
-    )
+    return initializeDatasources(datasourceMap, lens.datasourceStates, undefined, initialContext, {
+      isFullEditor: true,
+    })
       .then((result) => {
-        const datasourceStates = Object.entries(result).reduce(
-          (state, [datasourceId, datasourceState]) => ({
-            ...state,
-            [datasourceId]: {
-              ...datasourceState,
+        store.dispatch(
+          initEmpty({
+            newState: {
+              ...emptyState,
+              searchSessionId: currentSessionId || data.search.session.start(),
```

Comment:
this is what fixes the mentioned bug - somehow it was lost in the previous PRs and undefined searchSessionId caused infinite loop of requesting for data.